cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+                                componentNum, angleRefs.get(componentNum));
+                        LOG.info(log.get());
+                        continue;
+                    }
+                    angleRefs.put(componentNum, slackTerminal.getTerminal().getBusBreakerView().getBus().getId());
+                } else if (slackTerminal.getTerminal().getBusBreakerView().getBus() != null) {
+                    angleRefs.put(slackTerminal.getTerminal().getBusBreakerView().getBus().getId(),
+                            slackTerminal.getTerminal().getBusBreakerView().getBus().getId());
+                } else {
+                    Supplier<String> message = () -> String.format("Slack terminal at equipment %s is not connected and is not exported as slack terminal", slackTerminal.getTerminal().getConnectable().getId());
+                    LOG.info(message.get());
+                }
+            }
+        }
+        return angleRefs;
+    }
+
+    private static Map<String, List<String>> buildIslands(Network network) {
+        Map<String, List<String>> islands = new HashMap<>();
+        for (Bus b : network.getBusBreakerView().getBuses()) {
+            if (b.getSynchronousComponent() != null) {
+                int num = b.getSynchronousComponent().getNum();
+                islands.computeIfAbsent(String.valueOf(num), i -> new ArrayList<>());
+                islands.get(String.valueOf(num)).add(b.getId());
+            } else {
+                islands.put(b.getId(), Collections.singletonList(b.getId()));
+            }
+        }
+        return islands;
+    }
+
+    private static void writeVoltagesForTopologicalNodes(Network network, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        if (context.getTopologyKind() == CgmesTopologyKind.NODE_BREAKER) {
+            // TODO we need to export SV file data for NodeBraker
+            LOG.warn("NodeBreaker view require further investigation to map correctly Topological Nodes");
+            return;
+        }
+        for (Bus b : network.getBusBreakerView().getBuses()) {
+            writeVoltage(b.getId(), b.getV(), b.getAngle(), writer);
+        }
+    }
+
+    private static void writeVoltagesForBoundaryNodes(Network network, XMLStreamWriter writer) throws XMLStreamException {
+        for (DanglingLine dl : network.getDanglingLines()) {
+            Bus b = dl.getTerminal().getBusBreakerView().getBus();
+            Optional<String> topologicalNode = dl.getAliasFromType(CGMES_PREFIX_ALIAS + CgmesNames.TOPOLOGICAL_NODE);
+            if (topologicalNode.isPresent()) {
+                if (dl.hasProperty("v") && dl.hasProperty("angle")) {
+                    writeVoltage(topologicalNode.get(), Double.valueOf(dl.getProperty("v", "NaN")), Double.valueOf(dl.getProperty("angle", "NaN")), writer);
```

Comment:
I am not sure to get: what do you do if you have a NaN as v and angle? If you have that, I think that it could be nice to try to re-compute them as you do 2 lines below.

Reply:
I'm not sure if this is the place to recompute. We should ignore NaN values though, I'm modifying the code to do this.

Reply:
Edit: when the value is NaN, we put 0.0 as the CGMES convention requires.

Reply:
Ok !